chain-impl-mockchain/src/testing/e2e/transactions.rs

```diff
@@ -0,0 +1,91 @@
+use crate::{
+    fee::LinearFee,
+    testing::{
+        ledger::ConfigBuilder,
+        scenario::{prepare_scenario, wallet},
+        verifiers::LedgerStateVerifier,
+    },
+    value::Value,
+};
+use chain_addr::Discrimination;
+use quickcheck::TestResult;
+use quickcheck_macros::quickcheck;
+use rand::Rng;
+
+#[quickcheck]
+pub fn validate_ledger_state_after_transaction(amount: u64) -> TestResult {
+    if amount == 0 {
+        return TestResult::discard();
+    };
+
+    let mut rng = rand::thread_rng();
+
+    let (fee, total_fees) = (LinearFee::new(1, 1, 1), 3);
+    let (alice_initial_balance, bob_initial_balance) =
+        (amount + total_fees + 1, rng.gen_range(1..10));
+
+    let (mut ledger, controller) = prepare_scenario()
+        .with_config(
+            ConfigBuilder::new()
+                .with_discrimination(Discrimination::Test)
```

Comment:
is test discrimination important for this test case? If not, it's better to remove it, because reader can think that it has something to do with fees